AndroidTest/app/src/main/java/cloudant/com/androidtest/keyprovider/AndroidKeyProviderTests.java

```diff
@@ -0,0 +1,173 @@
+
+import com.cloudant.sync.datastore.encryption.AndroidKeyProvider;
+import com.cloudant.sync.datastore.encryption.DPKException;
+import com.cloudant.sync.datastore.encryption.EncryptionKey;
+import com.cloudant.sync.datastore.encryption.KeyManager;
+import com.cloudant.sync.datastore.encryption.KeyStorage;
+
+import java.util.Arrays;
+
+public class AndroidKeyProviderTests extends AndroidTestCase {
+
+    public void testCreateProviderWithIdentifier() {
+        AndroidKeyProvider provider = new AndroidKeyProvider(getContext(), "provider1password",
+                ProviderTestUtil.getUniqueIdentifier());
+
+        EncryptionKey createdKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not created", createdKey);
+
+        EncryptionKey loadedKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not loaded", loadedKey);
+
+        assertTrue("loadedKey did not match createdKey", Arrays.equals(createdKey.getKey(),
+                loadedKey.getKey()));
+    }
+
+    public void testCreateProviderWithManager() {
+        KeyStorage storage = new KeyStorage(getContext(), ProviderTestUtil.getUniqueIdentifier());
+        KeyManager manager = new KeyManager(storage);
+        AndroidKeyProvider provider = new AndroidKeyProvider("provider1password", manager);
+
+        EncryptionKey createdKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not created", createdKey);
+
+        EncryptionKey loadedKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not loaded", loadedKey);
+
+        assertTrue("loadedKey did not match createdKey", Arrays.equals(createdKey.getKey(),
+                loadedKey.getKey()));
+    }
+
+    public void testCreateTwoProvidersWithSameIdentifier() {
+        String identifier = ProviderTestUtil.getUniqueIdentifier();
+        AndroidKeyProvider provider1 = new AndroidKeyProvider(getContext(), "provider1password",
+                identifier);
+        AndroidKeyProvider provider2 = new AndroidKeyProvider(getContext(), "provider1password",
+                identifier);
+
+        EncryptionKey provider1Key = provider1.getEncryptionKey();
+        assertNotNull("provider1 EncryptionKey was not created", provider1Key);
+
+        EncryptionKey provider2Key = provider2.getEncryptionKey();
+        assertNotNull("provider2 EncryptionKey was not created", provider2Key);
+
+        assertTrue("provider1 key and provider2 key should be equal", Arrays.equals(provider1Key
+                .getKey(), provider2Key.getKey()));
+    }
+
+    public void testCreateTwoProvidersWithDifferentIdentifier() {
+        AndroidKeyProvider provider1 = new AndroidKeyProvider(getContext(), "provider1password",
+                ProviderTestUtil.getUniqueIdentifier());
+        AndroidKeyProvider provider2 = new AndroidKeyProvider(getContext(), "provider2password",
+                ProviderTestUtil.getUniqueIdentifier());
```

Comment:
Changes requested in comment included in https://github.com/cloudant/sync-android/commit/dd276d263bfc9c5ddebb047cecd27230cb074f63